### PR TITLE
Speed up tests

### PR DESCRIPTION
**Pull request by archiewood**
### Description

This PR aims to reduce the time it takes for our test suite to run.

It seems to cut test times to 25-30% of the original times.

Notable changes:
- Remove the pre `pnpm test` bundling (since this already happens in postinstall)
- Removes the post install rerun of `npm i`
- Removes the test env build completely (this was over 50% of our test time)

### Results

[Pre optimization test run](https://github.com/evidence-dev/evidence/actions/runs/8398470151/job/23003750927?pr=1780) 

**Ubuntu 18:** 6:44 -> 1:33
**Ubuntu 20:** 6:21 -> 1:34
**Mac 18:** 12mins -> 4mins
**Mac 20:** 10mins -> 4mins
**Windows 18:** 19mins -> 6min
**Windows 20:** 19mins -> 6min

### CODE REVIEW

**Comment by csjh** [comment]
I'm not too hot on dropping the `test-env` build test, unless it's replaced with a netlify build (which would be an improvement)

**Comment by archiewood** [comment]
Adding a netlify build is a great idea

**Comment by archiewood** [comment]
I added a netlify build but it fails due to OOM errors...

**Comment by archiewood** [comment]
boosted the memory and it worked

File: sites/test-env/package.json [hunk 1]
```diff
@@ -6,7 +6,6 @@
 		"build": "evidence build",
 		"build:strict": "evidence build:strict",
```
Comment (mcrascal) [inline sites/test-env/package.json:7]: update the build command to get it to go on netlify 
```suggestion
		"build:strict": "cross-env NODE_OPTIONS=--max-old-space-size=8192 vite build",

```

Comment (archiewood) [reply sites/test-env/package.json:7]: I actually inlined that into netlify, and the approach worked also - this seems cleaner though

why use vite build not evidence build? Does it matter

Comment (mcrascal) [reply sites/test-env/package.json:7]: sorry, I just copied and pasted from the example project, but it should be evidence build!

**Comment by archiewood** [comment]
closing in favour of https://github.com/evidence-dev/evidence/pull/1801 in order to trigger netlify build of test-env

**Pull request closed without merging**